apps/ledger-live-mobile/src/families/osmosis/operationDetails.js

```diff
@@ -0,0 +1,267 @@
+        v => v.validatorAddress === validator.address,
+      );
+
+      const formattedAmount = formatCurrencyUnit(
+        unit,
+        BigNumber(validator.amount),
+        {
+          disableRounding: true,
+          alwaysShowSign: false,
+          showCode: true,
+          discreet,
+          locale,
+        },
+      );
+
+      ret = (
+        <>
+          <Section
+            title={t("operationDetails.extra.delegatedTo")}
+            value={
+              formattedValidator?.name ??
+              formattedValidator?.validatorAddress ??
+              ""
+            }
+          />
+          <Section
+            title={t("operationDetails.extra.delegatedAmount")}
+            value={formattedAmount}
+          />
+          {autoClaimedRewards != null && autoClaimedRewards.gt(0) ? (
+            <Section
+              title={t("operationDetails.extra.autoClaimedRewards")}
+              value={formatCurrencyUnit(unit, autoClaimedRewards, formatConfig)}
+            />
+          ) : null}
+        </>
+      );
+      break;
+    }
+    case "UNDELEGATE": {
+      const { validators } = extra;
+      if (!validators || validators.length <= 0) break;
+
+      const validator = extra.validators[0];
+
+      const formattedValidator = cosmosValidators.find(
+        v => v.validatorAddress === validator.address,
+      );
+
+      const formattedAmount = formatCurrencyUnit(
+        unit,
+        BigNumber(validator.amount),
+        {
+          disableRounding: true,
+          alwaysShowSign: false,
+          showCode: true,
+          discreet,
+          locale,
+        },
+      );
+
+      ret = (
+        <>
+          <Section
+            title={t("operationDetails.extra.undelegatedFrom")}
+            value={formattedValidator?.name ?? validator.address}
+            onPress={() => {
+              redirectAddressCreator(validator.address);
+            }}
+          />
+          <Section
+            title={t("operationDetails.extra.undelegatedAmount")}
+            value={formattedAmount}
+          />
+          {autoClaimedRewards != null && autoClaimedRewards.gt(0) ? (
+            <Section
+              title={t("operationDetails.extra.autoClaimedRewards")}
+              value={formatCurrencyUnit(unit, autoClaimedRewards, formatConfig)}
+            />
+          ) : null}
+        </>
+      );
+      break;
+    }
+    case "REDELEGATE": {
+      const { sourceValidator, validators } = extra;
+      if (!validators || validators.length <= 0 || !sourceValidator) break;
+
+      const validator = extra.validators[0];
+
+      const formattedValidator = cosmosValidators.find(
+        v => v.validatorAddress === validator.address,
+      );
+
+      const formattedSourceValidator = cosmosValidators.find(
+        v => v.validatorAddress === sourceValidator,
+      );
+
+      const formattedAmount = formatCurrencyUnit(
+        unit,
+        BigNumber(validator.amount),
```

Comment:
it would be better not to have to parse at runtime but to have the amount already a BigNumber. isn't it already a BigNumber?

Reply:
same here: https://github.com/LedgerHQ/ledger-live/blob/c321c29c62d0ffcc7dd63d24e46444ba17e44c2d/apps/ledger-live-mobile/src/families/cosmos/operationDetails.js#L114.

Additionally, I looked up the implementation in LLD for both Osmosis and Cosmos and it looks like we cast to BigNumber in all of the above. If we want to change it, we should update it in all implementations for Cosmos, Osmosis and in both LLM and LLD, but unless it's deemed as significant I would abstain from making code changes at this stage (i.e. it might break stuff in some edge cases). Cc @JesseKuntz @trentkrogers